### DISCUSSION TITLE
Dot accessor wildcards

### DISCUSSION THREAD

**Discussion by FoolMuun** [schibsted/jslt#301] [Q&A] [answered]
In xpath you can use wildcards in the path in a Match expression - e.g. `/*/*/book` matches all "book" nodes located at that level in an XML hierarchy. Do we have a similar concept in JSLT to refer to nodes? Or am I just bad at using JSLT `for` loops?

E.g. input JSON
`[
{"BusbarSection" : {"Name" : "BB1", "FeederID" : "F1"}},
{"BusbarSection" : {"Name" : "BB2", "FeederID" : "F2"}},
{"BusbarSection" : {"Name" : "BB3", "FeederID" : "F3"}},
{"ShuntCompensator" : {"Name" : "SC1", "FeederID" : "F1"}},
{"ShuntCompensator" : {"Name" : "SC2", "FeederID" : "F2"}},
{"ShuntCompensator" : {"Name" : "SC3", "FeederID" : "F3"}}
]`

I can use this JSLT (which is kinda hard to read)
`{
  "EquipmentNames": [for (flatten([for (.) array(.)])) .value.Name]
}`

To produce this required result JSON
`{
  "EquipmentNames" : ["BB1", "BB2", "BB3", "SC1", "SC2", "SC3"  ]
}`

But the JSLT gets pretty rough if the data structure is nested deeper.
Is there something simpler like xpath's `/*/Name` that one can use to skip hierarchy levels e.g.
`for (..Name)`
?

**Comment by FoolMuun** [comment, marked as answer]
I 'spose what I'm looking for is the .. operator as proposed here [ideas/dotdot.md](https://github.com/schibsted/jslt/blob/master/ideas/dotdot.md)
+1 for that idea @larsga

**Reply by FoolMuun** [reply]
See also answer to this question: [access parent field in for expression](https://github.com/schibsted/jslt/discussions/163)

**Reply by larsga** [reply]
Thanks for the feedback! I do want to implement something similar to the dotdot operator.
